macros/multiple_databases/dateadd.sql

```diff
@@ -0,0 +1,22 @@
+{%- macro dateadd(datepart, number, datetime_field) -%}
+{%- if target.type == 'snowflake' -%}
+    dateadd({{ datepart }}, {{ number }}, try_to_timestamp(to_varchar({{ datetime_field }})))
+{%- elif target.type == 'sqlserver' -%}
+    dateadd({{ datepart }}, {{ number }}, try_convert(datetime2, {{ datetime_field }}))
+{%- elif target.type == 'databricks' -%}
+    {%- set clock_component -%}
+        to_unix_timestamp({{ datetime_field }}) - to_unix_timestamp(date_trunc('DD', {{ datetime_field }}))
```

Comment:
If we convert the input value using to_unix_timestamp, we will lose the millisecond precision. This makes it different from the Snowflake and SQL Server implementations